Webclient: misc improvements (was: parse html in server message)

ctrlaltca opened the pull request:
Fix #2088

Implements an htmlSanitize() method using a whitelist approach to only allows some specific html tags, attributes and protocols in anchors.

Minor: added favicon and clientfeatures set needed to connect to wooger's server

tooomm commented on webclient/webclient.js:
```diff
@@ -158,7 +158,7 @@ var WebClient = {
       "clientid" : this.guid(),
       "clientver" : "webclient-0.1 (2015-12-23)",
-      "clientfeatures" : [ "client_id", "client_ver"],
+      "clientfeatures" : [ "client_id", "client_ver", "client_warnings", "feature_set", "mod_log_lookup", "room_chat_history", "user_ban_history"]
```
Does the webclient support all these??

Daenyth replied:
I don't think so

ctrlaltca replied:
i had to add them in order to make the client able to connect to wooger's server, that requires some of those  features in clients (i just added them all, didn't check them one by one).

tooomm replied:
So they shouldn't be listed?

tooomm commented:
>Minor: added (...) clientfeatures set needed to connect to wooger's server

You can connect to http://cockatrice.woogerworks.com/play/ right now?
In the `Account` tab you see this:
![missing_features](https://cloud.githubusercontent.com/assets/9874850/17247833/4f7d47ae-5596-11e6-83e7-dea89cac9d41.png)

Daenyth commented:
I don't think those are mandatory features, just suggested. @woogerboy21  ?

tooomm commented:
I didn't know that we differentiate between suggested and requiered!?

ZeldaZach commented on webclient/index.html:
```diff
@@ -26,7 +27,7 @@
     <h3>Login to server</h3>  
     <label for="host">Host</label>
-    <input type="text" id="host" value="127.0.0.1" />
+    <input type="text" id="host" value="cockatrice.woogerworks.com" />
```
Do we want this, or was this a test you were doing?

ctrlaltca replied:
Which one is a better default? 127.0.0.1 is probably good only for local testing

woogerboy21 replied:
I didnt look, but if your going to expand the pre-defined host string you may want to expand the box character length as well. I know when I worked on the page at the woogerworks site I expanded the box length to fit the entire string without causing you to have to arrow over to see the right most characters.

woogerboy21 commented:
IMO.  The code should not be adapted to fit one particular host over another.  I do understand why some of the things set with this change are the way they are since woogerworks is the most widely used host.  The question I have would be, how difficult is it for the provider of the service (web client in this case) to make adjustments for there preferences in order to make the web client work?  I already went through on the version I am hosting and made the changes rather easy but would there be an expectation for some one else to do so?

woogerboy21 commented:
I dont know if i like the idea of putting features into a client that doesnt really support the features.

woogerboy21 commented:
> I don't think those are mandatory features, just suggested. @woogerboy21 ?

The only two features I have set as required are: client_ver,client_warnings

It would be nice to see the web client have the ability to do client warnings at some point.  Only because right now some one could log into the web client, spew crap, a mod warn them and they would never see the warning.

ctrlaltca commented:
I'll update the PR adding  real support for client_warnings, removing the unneeded client features and implementing all the other suggested changes.

ctrlaltca commented:
Implemented changes:
 * Add an autocomplete with known server names to the connect dialog;
 * Added support for server messages (identification, shutdown, user warning and promotion);
 * Connection dialog’s inputs are now wider;
 * Only declare implemented client features;
 * Added check for protocol version on connect before login.
